### ISSUE TITLE
Bug: `// eslint-disable-next-line complexity` doesn't disable linting on decorated method

### ISSUE THREAD

**Issue by AndreyWV** [eslint/eslint#20357]
### Environment

Node version: v22.16.0
npm version: v10.9.2
Local ESLint version: v9.39.1 (Currently used)
Global ESLint version: Not found
Operating System: linux 6.6.87.2-microsoft-standard-WSL2
---
Angular version: 21.0.0

### What parser are you using?

@typescript-eslint/parser

### What did you do?

<details>
<summary>Configuration</summary>

```js
import globals from 'globals';
import pluginJs from '@eslint/js';
import tseslint from 'typescript-eslint';
import angularTemplate from '@angular-eslint/eslint-plugin-template';
import angularTemplateParser from '@angular-eslint/template-parser';

/** @type {import('eslint').Linter.Config[]} */
export default [
  pluginJs.configs.recommended,
  ...tseslint.configs.recommended,
  {
    files: [
      'src/**/*.ts',
    ],
    languageOptions: {
      globals: globals.browser,
      parserOptions: {
        project: ['tsconfig.json'],
      },
    },
    rules: {
      '@typescript-eslint/explicit-function-return-type': [
        'error',
        {
          allowIIFEs: true,
        },
      ],

      '@typescript-eslint/no-explicit-any': [
        'warn',
        {
          ignoreRestArgs: true,
          fixToUnknown: false,
        },
      ],

      'max-len': [
        'error',
        {
          code: 120,
          ignoreStrings: true,
          ignoreUrls: true,
          ignoreRegExpLiterals: true,
          tabWidth: 2,
        },
      ],

      'operator-linebreak': [
        'error',
        'before',
      ],

      'multiline-ternary': [
        'error',
        'always',
      ],

      'object-curly-spacing': [
        'error',
        'always',
      ],

      quotes: [
        'error',
        'single',
      ],

      semi: [
        'error',
        'always',
      ],

      'no-console': [
        'error',
      ],

      'comma-dangle': [
        'error',
        'always-multiline',
      ],

      complexity: [
        'error',
        {
          max: 6,
        },
      ],

      '@typescript-eslint/no-shadow': 'error',
    },
  },
  {
    files: [
      'src/**/*.html',
    ],
    languageOptions: {
      parser: angularTemplateParser,
      globals: globals.browser,
    },
    plugins: {
      '@angular-eslint/template': angularTemplate,
    },
    rules: {
      ...angularTemplate.configs.recommended.rules,
      '@typescript-eslint/ban-ts-comment': 'off',
    },
  },
];

```
</details>

```js
// src/main.ts
function testDecorator(): MethodDecorator {
  return (_: unknown, __: unknown, descriptor: PropertyDescriptor): PropertyDescriptor => {
    return descriptor;
  };
}

export class TestComponent {

  @testDecorator()
  // eslint-disable-next-line complexity
  public async testMethodDecorated(): Promise<void> {
    let a = 1;
    a++;
    if (a > 2) {
      // empty
    } else if (a > 3) {
      // empty
    } else if (a > 4) {
      // empty
    } else if (a > 5) {
      // empty
    } else if (a > 6) {
      // empty
    } else if (a > 7) {
      // empty
    }
  }

  // eslint-disable-next-line complexity
  public testMethod(): void {
    let a = 1;
    a++;
    if (a > 2) {
      // empty
    } else if (a > 3) {
      // empty
    } else if (a > 4) {
      // empty
    } else if (a > 5) {
      // empty
    } else if (a > 6) {
      // empty
    } else if (a > 7) {
      // empty
    }
  }

}
```

### What did you expect to happen?

```
> All files pass linting.
```
Ignoring the rules should work

### What actually happened?

```
/home/projects/stackblitz-starters-fdrhel83/src/main.ts
   9:3  error    Method 'testMethodDecorated' has a complexity of 7. Maximum allowed is 6       complexity
  10:3  warning  Unused eslint-disable directive (no problems were reported from 'complexity')

✖ 2 problems (1 error, 1 warning)
  0 errors and 1 warning potentially fixable with the `--fix` option.

Lint warnings found in the listed files.

Lint errors found in the listed files.
```

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/stackblitz-starters-fdrhel83?file=src%2Fmain.ts

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

`ng lint` and `eslint` leads to same error.

**Comment by DMartens** [comment]
I think this does not work as the decorator is part of the ignored AST node even though it is before the method (quirk of the AST).
The eslint ignore directive works when you move it above the decorator ([modified stackblitz](https://stackblitz.com/edit/stackblitz-starters-og1tlhqp?file=src%2Fmain.ts)):
```typescript
// eslint-disable-next-line complexity
@testDecorator()
public async testMethodDecorated(): Promise<void> {}
```

**Comment by AndreyWV** [comment]
@DMartens Agree, behavior changed after laast update. Thanks
